docs/synapse/userguides/storm_ref_filter.rstorm

```diff
@@ -793,247 +917,449 @@ know the exact order or values of the array itself.
 
-- Filter results to exclude nodes tagged as sinkholes:
+- ``rep.microsoft.aqua_blizzard``
+- ``rep.microsoft.cadet_blizzard``
+- ``rep.microsoft.forest_blizzard``
+- ``rep.microsoft.very.long.tag.thatendswithblizzard``
 
+...etc.
 
-.. storm-pre:: inet:ipv4 -#cno.infra.dns.sinkhole
+*Filter the current working set to exclude any nodes tagged with any tag that starts with "cno" and is followed by any string:*
+
+.. storm-pre:: [ ( inet:ipv4=93.90.223.185 +#cno.infra.dns.sink.hole ) ( inet:fqdn=vertex.link +#cno ) ] -#cno**
 ::
-    
-    <query> -#cno.infra.dns.sinkhole
+  
+  <query> -#cno**
+
+The tag glob filter above uses a double asterisk to match any string following "cno". The filter will match all of
+the following:
 
+- ``cno.mal``
+- ``cno.threat.t42``
+- ``cnoooo.you_get_a_cno.and_you_get_a_cno``
 
-*Filter by tag and time:*
+...etc.
 
-- Filter results to include only nodes that were associated with TOR infrastructure as of December 12, 2019:
+.. NOTE::
+  
+  The double asterisk must match "something" - the filter above will **not** match a node that simply has the tag ``#cno``.
 
 
-.. storm-pre:: inet:fqdn +#cno.infra.anon.tor@=2019/12/12
+*Filter the current working set to include any nodes tagged with any "APT" group by any third-party reporting organization:*
+
+.. storm-pre:: [ ( inet:fqdn=www.autoapec.com +#rep.alienvault.apt30 ) ( inet:fqdn=intuitstatistic.com +#rep.mandiant.apt28 ) ( inet:fqdn=supportus.mefound.com +#rep.pwc.apt10 ) ] +#rep.*.apt**
 ::
-    
-    <query> +#cno.infra.anon.tor@=2019/12/12
+  
+  <query> +#rep.*.apt**
 
+The tag glob filter above uses both a single and double asterisk. The single asterisk matches any tag element in the second
+position; the double asterisk matches any string following "apt". The filter will match all of the following:
 
-*Filter by tag and time interval:*
+- ``rep.alienvault.apt30``
+- ``rep.mandiant.apt28``
+- ``rep.pwc.apt10``
+- ``rep.malwarebazaar.apt_malware_exe_dll_ps1``
 
-- Filter results to include only those nodes associated with sinkhole infrastructure between January 1,
-  2017 and January 1, 2018:
+...etc.
 
 
-.. storm-pre:: [inet:fqdn=woot.com inet:fqdn=vertex.link inet:fqdn=google.com]
+.. _filter-tag-timestamp:
+
+Filter Using Tag Timestamp Values
++++++++++++++++++++++++++++++++++
+
+A tag timestamp can be thought of as a specialized "property" of a tag that happens to be a date / time range
+(interval). You can filter nodes based on tag timestamp values using any comparison operator supported by interval
+(:ref:`type-ival` types). The time / interval extended operator ( ``@=`` ) is used most often, but equal to
+( ``=`` ) can also be used to **exactly** match the values in the interval.
+
+See :ref:`filter-interval` for additional detail on the use of the ``@=`` operator.
+
+.. NOTE::
+  
+  It is not currently possible to filter generically based on "tags that have timestamps" vs. "tags that do not
```

Comment:
You can do something like `+$((#rep.foo).0=$lib.null)` but I'm assuming that type of filter doesn't quite fit in here.

Reply:
Updated. Added your expression as well as the "analyst's hack" method - plz lmk if you think that should be removed (I'm good either way).

Reply:
I think your hack method is fine, it is probably actually better since in some cases that can be used to optimize the lift.